freqtrade/strategy/interface.py

```diff
@@ -132,14 +132,18 @@ def __init__(self, config: dict) -> None:
         self._last_candle_seen_per_pair: Dict[str, datetime] = {}
         super().__init__(config)
 
-    @abstractmethod
     def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
         """
         Populate indicators that will be used in the Buy and Sell strategy
         :param dataframe: DataFrame with data from the exchange
         :param metadata: Additional information, like the currently traded pair
         :return: a Dataframe with all mandatory indicators for the strategies
         """
+        for fn in [getattr(self.__class__, func) for func in dir(self.__class__)
+                   if callable(getattr(self.__class__, func))]:
+            if not getattr(fn, '_is_informative', False):
+                continue
+            dataframe = fn(self, dataframe, metadata)
```

Comment:
it's not clear to me what you're trying to do here - but it's definitely not a good idea to do it this way.

It assumes that you'll not have a `populate_indicators` method - which will only be confusing for existing users, or new users using an available method.
This is an abstract method (for a reason, really) - and the current documentation / samples / available strategies do not call the parent method - so the "working" of this will be fluky at best.

If you think this is a neccessity (i'm not sure - but it's not clear to me what this does just yet) - then keep `populate_indicators` as abstract method - and write a wrapper (or use the existing wrapper in `advise_indicators`) for this task.

Reply:
> it's not clear to me what you're trying to do here

This code iterates all callables of the class and calls ones tagged with `_is_informative = True`.

> It assumes that you'll not have a populate_indicators method - which will only be confusing for existing users, or new users using an available method.

That is not true. `populate_indicators` is still required just like it used to be.

> This is an abstract method (for a reason, really)

You have a point about loss of `@abstract` decorator. Can we keep it to mandate implementation of such method, while also calling this method via `super()`?

> current documentation / samples / available strategies do not call the parent method

They do not. And it is not required, unless decorator is used. Maybe we could side-step this requirement by calling this machinery before invoking `populate_indicators`. It would allow keeping `@abstract` too.

> If you think this is a neccessity (i'm not sure - but it's not clear to me what this does just yet) - then keep populate_indicators as abstract method - and write a wrapper (or use the existing wrapper in advise_indicators) for this task.

Yes, most likely best course of action 👍🏻